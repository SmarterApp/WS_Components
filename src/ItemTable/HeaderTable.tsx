import * as React from "react";
import {
  SortColumnModel,
  HeaderSortModel,
  SortDirection
} from "./ItemTableModels";

/**
 * Properties for HeaderTable component
 * @interface HeaderTableProps
 */
export interface HeaderTableProps {
  columns: SortColumnModel[];
  onHeaderClick: (header: SortColumnModel) => void;
  sorts: HeaderSortModel[];
}

const invokeResetSortLimit = 2;

const style = {
  color: "gray"
};

const descendingArrow = (
  <span style={style} className="fa fa-sort-desc" aria-hidden="true" />
);

const ascendingArrow = (
  <span style={style} className="fa fa-sort-asc" aria-hidden="true" />
);

const noSort = <span style={style} className="fa fa-sort" aria-hidden="true" />;

/**
 * HeaderTable creates a table header based on the passed in columns
 * The HeaderTable, when clicked, will add the clicked column header
 * the parent, ItemTableContainer, state, sorting the table
 */
export class HeaderTable extends React.Component<HeaderTableProps, {}> {
  constructor(props: HeaderTableProps) {
    super(props);
    this.state = {
      dirElem: undefined
    };
  }
  /**
   * Send the clicked sort column to the parent to be added to the sorts list
   * using onHeaderClick prop.
   * @param {SortColumnModel} sCol
   * @param {(HeaderSortModel | undefined)} hCol
   */
  headerEventHandler(sCol: SortColumnModel, hCol: HeaderSortModel | undefined) {
    this.props.onHeaderClick(sCol);
  }

  /**
   * Assigns an ascending or descending arrow character to the visible column
   * header when clicked, denoting how it is sorted
   * @param {(HeaderSortModel | undefined)} headerSort
   * @returns {JSX.Element}
   */
  setDirElem(headerSort: HeaderSortModel | undefined): JSX.Element {
    let dirElem = noSort;
    if (!headerSort) {
      return dirElem;
    } else if (headerSort.direction === SortDirection.Ascending) {
      dirElem = ascendingArrow;
    } else if (headerSort.direction === SortDirection.Descending) {
      dirElem = descendingArrow;
    }
    return dirElem;
  }

  /**
   * Renders a single table header element and the corresponding ascending, descending,
   * or 'not-sorted' symbol
   * @param {SortColumnModel} col
   * @returns {JSX.Element}
   */
  renderHeader(col: SortColumnModel): JSX.Element {
    const headerSort = this.props.sorts.find(
      hs => hs.col.header === col.header
    );
    if (headerSort) {
      this.setDirElem(headerSort);
    }

    return (
      <th
        key={col.header}
        className={col.className}
        onClick={() => this.headerEventHandler(col, headerSort)}
      >
        <div className={col.className}>
          <span>{col.header}</span>
          {this.setDirElem(headerSort)}
        </div>
      </th>
    );
  }

  render() {
    return (
      <thead>
        <tr className="primary">
          <th />
          {this.props.columns.map(col => this.renderHeader(col))}
        </tr>
      </thead>
    );
  }
}
