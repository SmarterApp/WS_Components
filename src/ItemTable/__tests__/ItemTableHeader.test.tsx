import * as React from "react";
import * as ReactDOM from "react-dom";
import {
  headerColumns,
  HeaderSortModel,
  SortColumnModel,
  SortDirection
} from "../ItemTableModels";
import * as TestUtils from "react-dom/test-utils";
import { shallow, mount, render } from "enzyme";
import { HeaderTable } from "../HeaderTable";

describe("ItemTableHeader", () => {
  const tabs = [
    "item",
    "claimAndTarget",
    "subject",
    "grade",
    "interactionType"
  ];

  const sorts: Array<HeaderSortModel> = [
    {
      col: headerColumns[0],
      direction: SortDirection.Ascending,
      resetSortCount: 1
    }
  ];

  const props = {
    columns: headerColumns,
    onHeaderClick: jest.fn((header: SortColumnModel) => null),
    sorts
  };

  it("matches snapshot", () => {
    expect(shallow(<HeaderTable {...props} />)).toMatchSnapshot();
  });

  it("calls onHeaderClick", () => {
    let wrapper = shallow(<HeaderTable {...props} />);
    tabs.forEach(tab => {
      wrapper.find(`th.${tab}`).simulate("click");
      expect(props.onHeaderClick).toHaveBeenCalled();
    });
  });

  it("sorts list on header click", () => {
    let wrapper = shallow(<HeaderTable {...props} />);
    tabs.forEach(tab => {
      wrapper.find(`th.${tab}`).simulate("click");
      expect(JSON.stringify(wrapper.html())).toMatchSnapshot();
      wrapper.find(`th.${tab}`).simulate("click");
      expect(JSON.stringify(wrapper.html())).toMatchSnapshot();
      wrapper.find(`th.${tab}`).simulate("click");
      expect(JSON.stringify(wrapper.html())).toMatchSnapshot();
    });
  });
});
