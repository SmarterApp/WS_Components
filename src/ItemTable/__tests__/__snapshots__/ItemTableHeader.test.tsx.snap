// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemTableHeader matches snapshot 1`] = `
<thead>
  <tr
    className="primary"
  >
    <th
      colSpan={2}
    />
    <th
      className="item"
      key="Item"
      onClick={[Function]}
    >
      <div
        className="item"
      >
        <span>
          Item
        </span>
        <span
          aria-hidden="true"
          className="fa fa-sort-asc"
          style={
            Object {
              "color": "gray",
            }
          }
        />
      </div>
    </th>
    <th
      className="claimAndTarget"
      key="Claim/Target"
      onClick={[Function]}
    >
      <div
        className="claimAndTarget"
      >
        <span>
          Claim/Target
        </span>
        <span
          aria-hidden="true"
          className="fa fa-sort"
          style={
            Object {
              "color": "gray",
            }
          }
        />
      </div>
    </th>
    <th
      className="subject"
      key="Subject"
      onClick={[Function]}
    >
      <div
        className="subject"
      >
        <span>
          Subject
        </span>
        <span
          aria-hidden="true"
          className="fa fa-sort"
          style={
            Object {
              "color": "gray",
            }
          }
        />
      </div>
    </th>
    <th
      className="grade"
      key="Grade"
      onClick={[Function]}
    >
      <div
        className="grade"
      >
        <span>
          Grade
        </span>
        <span
          aria-hidden="true"
          className="fa fa-sort"
          style={
            Object {
              "color": "gray",
            }
          }
        />
      </div>
    </th>
    <th
      className="interactionType"
      key="Item Type"
      onClick={[Function]}
    >
      <div
        className="interactionType"
      >
        <span>
          Item Type
        </span>
        <span
          aria-hidden="true"
          className="fa fa-sort"
          style={
            Object {
              "color": "gray",
            }
          }
        />
      </div>
    </th>
  </tr>
</thead>
`;

exports[`ItemTableHeader sorts list on header click 1`] = `"\\"<thead><tr class=\\\\\\"primary\\\\\\"><th colSpan=\\\\\\"2\\\\\\"></th><th class=\\\\\\"item\\\\\\"><div class=\\\\\\"item\\\\\\"><span>Item</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort-asc\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"claimAndTarget\\\\\\"><div class=\\\\\\"claimAndTarget\\\\\\"><span>Claim/Target</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"subject\\\\\\"><div class=\\\\\\"subject\\\\\\"><span>Subject</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"grade\\\\\\"><div class=\\\\\\"grade\\\\\\"><span>Grade</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"interactionType\\\\\\"><div class=\\\\\\"interactionType\\\\\\"><span>Item Type</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th></tr></thead>\\""`;

exports[`ItemTableHeader sorts list on header click 2`] = `"\\"<thead><tr class=\\\\\\"primary\\\\\\"><th colSpan=\\\\\\"2\\\\\\"></th><th class=\\\\\\"item\\\\\\"><div class=\\\\\\"item\\\\\\"><span>Item</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort-asc\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"claimAndTarget\\\\\\"><div class=\\\\\\"claimAndTarget\\\\\\"><span>Claim/Target</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"subject\\\\\\"><div class=\\\\\\"subject\\\\\\"><span>Subject</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"grade\\\\\\"><div class=\\\\\\"grade\\\\\\"><span>Grade</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"interactionType\\\\\\"><div class=\\\\\\"interactionType\\\\\\"><span>Item Type</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th></tr></thead>\\""`;

exports[`ItemTableHeader sorts list on header click 3`] = `"\\"<thead><tr class=\\\\\\"primary\\\\\\"><th colSpan=\\\\\\"2\\\\\\"></th><th class=\\\\\\"item\\\\\\"><div class=\\\\\\"item\\\\\\"><span>Item</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort-asc\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"claimAndTarget\\\\\\"><div class=\\\\\\"claimAndTarget\\\\\\"><span>Claim/Target</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"subject\\\\\\"><div class=\\\\\\"subject\\\\\\"><span>Subject</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"grade\\\\\\"><div class=\\\\\\"grade\\\\\\"><span>Grade</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"interactionType\\\\\\"><div class=\\\\\\"interactionType\\\\\\"><span>Item Type</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th></tr></thead>\\""`;

exports[`ItemTableHeader sorts list on header click 4`] = `"\\"<thead><tr class=\\\\\\"primary\\\\\\"><th colSpan=\\\\\\"2\\\\\\"></th><th class=\\\\\\"item\\\\\\"><div class=\\\\\\"item\\\\\\"><span>Item</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort-asc\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"claimAndTarget\\\\\\"><div class=\\\\\\"claimAndTarget\\\\\\"><span>Claim/Target</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"subject\\\\\\"><div class=\\\\\\"subject\\\\\\"><span>Subject</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"grade\\\\\\"><div class=\\\\\\"grade\\\\\\"><span>Grade</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"interactionType\\\\\\"><div class=\\\\\\"interactionType\\\\\\"><span>Item Type</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th></tr></thead>\\""`;

exports[`ItemTableHeader sorts list on header click 5`] = `"\\"<thead><tr class=\\\\\\"primary\\\\\\"><th colSpan=\\\\\\"2\\\\\\"></th><th class=\\\\\\"item\\\\\\"><div class=\\\\\\"item\\\\\\"><span>Item</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort-asc\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"claimAndTarget\\\\\\"><div class=\\\\\\"claimAndTarget\\\\\\"><span>Claim/Target</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"subject\\\\\\"><div class=\\\\\\"subject\\\\\\"><span>Subject</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"grade\\\\\\"><div class=\\\\\\"grade\\\\\\"><span>Grade</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"interactionType\\\\\\"><div class=\\\\\\"interactionType\\\\\\"><span>Item Type</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th></tr></thead>\\""`;

exports[`ItemTableHeader sorts list on header click 6`] = `"\\"<thead><tr class=\\\\\\"primary\\\\\\"><th colSpan=\\\\\\"2\\\\\\"></th><th class=\\\\\\"item\\\\\\"><div class=\\\\\\"item\\\\\\"><span>Item</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort-asc\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"claimAndTarget\\\\\\"><div class=\\\\\\"claimAndTarget\\\\\\"><span>Claim/Target</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"subject\\\\\\"><div class=\\\\\\"subject\\\\\\"><span>Subject</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"grade\\\\\\"><div class=\\\\\\"grade\\\\\\"><span>Grade</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"interactionType\\\\\\"><div class=\\\\\\"interactionType\\\\\\"><span>Item Type</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th></tr></thead>\\""`;

exports[`ItemTableHeader sorts list on header click 7`] = `"\\"<thead><tr class=\\\\\\"primary\\\\\\"><th colSpan=\\\\\\"2\\\\\\"></th><th class=\\\\\\"item\\\\\\"><div class=\\\\\\"item\\\\\\"><span>Item</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort-asc\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"claimAndTarget\\\\\\"><div class=\\\\\\"claimAndTarget\\\\\\"><span>Claim/Target</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"subject\\\\\\"><div class=\\\\\\"subject\\\\\\"><span>Subject</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"grade\\\\\\"><div class=\\\\\\"grade\\\\\\"><span>Grade</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"interactionType\\\\\\"><div class=\\\\\\"interactionType\\\\\\"><span>Item Type</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th></tr></thead>\\""`;

exports[`ItemTableHeader sorts list on header click 8`] = `"\\"<thead><tr class=\\\\\\"primary\\\\\\"><th colSpan=\\\\\\"2\\\\\\"></th><th class=\\\\\\"item\\\\\\"><div class=\\\\\\"item\\\\\\"><span>Item</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort-asc\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"claimAndTarget\\\\\\"><div class=\\\\\\"claimAndTarget\\\\\\"><span>Claim/Target</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"subject\\\\\\"><div class=\\\\\\"subject\\\\\\"><span>Subject</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"grade\\\\\\"><div class=\\\\\\"grade\\\\\\"><span>Grade</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"interactionType\\\\\\"><div class=\\\\\\"interactionType\\\\\\"><span>Item Type</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th></tr></thead>\\""`;

exports[`ItemTableHeader sorts list on header click 9`] = `"\\"<thead><tr class=\\\\\\"primary\\\\\\"><th colSpan=\\\\\\"2\\\\\\"></th><th class=\\\\\\"item\\\\\\"><div class=\\\\\\"item\\\\\\"><span>Item</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort-asc\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"claimAndTarget\\\\\\"><div class=\\\\\\"claimAndTarget\\\\\\"><span>Claim/Target</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"subject\\\\\\"><div class=\\\\\\"subject\\\\\\"><span>Subject</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"grade\\\\\\"><div class=\\\\\\"grade\\\\\\"><span>Grade</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"interactionType\\\\\\"><div class=\\\\\\"interactionType\\\\\\"><span>Item Type</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th></tr></thead>\\""`;

exports[`ItemTableHeader sorts list on header click 10`] = `"\\"<thead><tr class=\\\\\\"primary\\\\\\"><th colSpan=\\\\\\"2\\\\\\"></th><th class=\\\\\\"item\\\\\\"><div class=\\\\\\"item\\\\\\"><span>Item</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort-asc\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"claimAndTarget\\\\\\"><div class=\\\\\\"claimAndTarget\\\\\\"><span>Claim/Target</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"subject\\\\\\"><div class=\\\\\\"subject\\\\\\"><span>Subject</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"grade\\\\\\"><div class=\\\\\\"grade\\\\\\"><span>Grade</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"interactionType\\\\\\"><div class=\\\\\\"interactionType\\\\\\"><span>Item Type</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th></tr></thead>\\""`;

exports[`ItemTableHeader sorts list on header click 11`] = `"\\"<thead><tr class=\\\\\\"primary\\\\\\"><th colSpan=\\\\\\"2\\\\\\"></th><th class=\\\\\\"item\\\\\\"><div class=\\\\\\"item\\\\\\"><span>Item</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort-asc\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"claimAndTarget\\\\\\"><div class=\\\\\\"claimAndTarget\\\\\\"><span>Claim/Target</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"subject\\\\\\"><div class=\\\\\\"subject\\\\\\"><span>Subject</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"grade\\\\\\"><div class=\\\\\\"grade\\\\\\"><span>Grade</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"interactionType\\\\\\"><div class=\\\\\\"interactionType\\\\\\"><span>Item Type</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th></tr></thead>\\""`;

exports[`ItemTableHeader sorts list on header click 12`] = `"\\"<thead><tr class=\\\\\\"primary\\\\\\"><th colSpan=\\\\\\"2\\\\\\"></th><th class=\\\\\\"item\\\\\\"><div class=\\\\\\"item\\\\\\"><span>Item</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort-asc\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"claimAndTarget\\\\\\"><div class=\\\\\\"claimAndTarget\\\\\\"><span>Claim/Target</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"subject\\\\\\"><div class=\\\\\\"subject\\\\\\"><span>Subject</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"grade\\\\\\"><div class=\\\\\\"grade\\\\\\"><span>Grade</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"interactionType\\\\\\"><div class=\\\\\\"interactionType\\\\\\"><span>Item Type</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th></tr></thead>\\""`;

exports[`ItemTableHeader sorts list on header click 13`] = `"\\"<thead><tr class=\\\\\\"primary\\\\\\"><th colSpan=\\\\\\"2\\\\\\"></th><th class=\\\\\\"item\\\\\\"><div class=\\\\\\"item\\\\\\"><span>Item</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort-asc\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"claimAndTarget\\\\\\"><div class=\\\\\\"claimAndTarget\\\\\\"><span>Claim/Target</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"subject\\\\\\"><div class=\\\\\\"subject\\\\\\"><span>Subject</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"grade\\\\\\"><div class=\\\\\\"grade\\\\\\"><span>Grade</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"interactionType\\\\\\"><div class=\\\\\\"interactionType\\\\\\"><span>Item Type</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th></tr></thead>\\""`;

exports[`ItemTableHeader sorts list on header click 14`] = `"\\"<thead><tr class=\\\\\\"primary\\\\\\"><th colSpan=\\\\\\"2\\\\\\"></th><th class=\\\\\\"item\\\\\\"><div class=\\\\\\"item\\\\\\"><span>Item</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort-asc\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"claimAndTarget\\\\\\"><div class=\\\\\\"claimAndTarget\\\\\\"><span>Claim/Target</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"subject\\\\\\"><div class=\\\\\\"subject\\\\\\"><span>Subject</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"grade\\\\\\"><div class=\\\\\\"grade\\\\\\"><span>Grade</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"interactionType\\\\\\"><div class=\\\\\\"interactionType\\\\\\"><span>Item Type</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th></tr></thead>\\""`;

exports[`ItemTableHeader sorts list on header click 15`] = `"\\"<thead><tr class=\\\\\\"primary\\\\\\"><th colSpan=\\\\\\"2\\\\\\"></th><th class=\\\\\\"item\\\\\\"><div class=\\\\\\"item\\\\\\"><span>Item</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort-asc\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"claimAndTarget\\\\\\"><div class=\\\\\\"claimAndTarget\\\\\\"><span>Claim/Target</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"subject\\\\\\"><div class=\\\\\\"subject\\\\\\"><span>Subject</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"grade\\\\\\"><div class=\\\\\\"grade\\\\\\"><span>Grade</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th><th class=\\\\\\"interactionType\\\\\\"><div class=\\\\\\"interactionType\\\\\\"><span>Item Type</span><span style=\\\\\\"color:gray\\\\\\" class=\\\\\\"fa fa-sort\\\\\\" aria-hidden=\\\\\\"true\\\\\\"></span></div></th></tr></thead>\\""`;
